exercises/protein-translation/.meta/src/reference/java/ProteinTranslator.java

```diff
@@ -0,0 +1,47 @@
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+class ProteinTranslator {
+
+    private static final String[] METHIONINE = {"AUG"};
+    private static final String[] PHENYLALANINE = {"UUU", "UUC"};
+    private static final String[] LEUCINE = {"UUA", "UUG"};
+    private static final String[] SERINE = {"UCU", "UCC", "UCA", "UCG"};
+    private static final String[] TYROSINE = {"UAU", "UAC"};
+    private static final String[] CYSTEINE = {"UGU", "UGC"};
+    private static final String[] TRYPTOPHAN = {"UGG"};
+    private static final String[] STOP = {"UAA", "UAG", "UGA"};
+
+    List<String> translate (String rnaSequence) {
+
+        List proteinList = new ArrayList();
```

Comment:
This should be `List<String> proteinList = new ArrayList<String>()`. Otherwise you're using [a list of generic `Object`s](https://www.ntu.edu.sg/home/ehchua/programming/java/JavaGeneric.html) which isn't [typesafe](https://stackoverflow.com/questions/260626/what-is-type-safe). Does that make sense? :)